internal/impl/ollama/base_processor.go

```diff
@@ -55,22 +65,111 @@ func (co *commandOutput) Write(b []byte) (int, error) {
 
 type key int
 
-const loggerKey key = key(1)
+const (
+	configKey key = key(1)
+)
+
+type runOllamaConfig struct {
+	logger      *service.Logger
+	fs          *service.FS
+	cacheDir    string
+	downloadURL string
+}
+
+// lookPath is very similar to exec.LookPath, except that it uses service.FS filesystem
+// abstractions.
+func (c *runOllamaConfig) lookPath(file string) (string, error) {
+	if runtime.GOOS != "linux" && runtime.GOOS != "darwin" {
+		return "", errors.ErrUnsupported
+	}
+	path := os.Getenv("PATH")
+	for _, dir := range filepath.SplitList(path) {
+		if !filepath.IsAbs(path) {
+			continue
+		}
+		path := filepath.Join(dir, file)
+		p, err := c.fs.Stat(path)
+		if err != nil || p.IsDir() {
+			continue
+		}
+		// Check that the file is executable
+		if err := syscall.Access(path, 0x1); err == nil {
+			return path, nil
+		}
+	}
+	return "", exec.ErrNotFound
+}
+
+func (c *runOllamaConfig) downloadOllama(ctx context.Context, path string) error {
+	var url string
+	const baseURL string = "https://github.com/ollama/ollama/releases/download/v0.3.6/ollama"
+	switch runtime.GOOS {
+	case "darwin":
+		// They ship an universal executable for darwin
+		url = baseURL + "-darwin"
+	case "linux":
+		url = fmt.Sprintf("%s-%s-%s", baseURL, runtime.GOOS, runtime.GOARCH)
+	default:
+		return fmt.Errorf("automatic download of ollama is not supported on %s, please download ollama manually", runtime.GOOS)
+	}
+	if c.downloadURL != "" {
```

Comment:
Should we maybe move this check above the switch just in case people have a download link for Windows?

Reply:
Oh good catch! Fixed.